.github/get-latest-index.py

```diff
@@ -0,0 +1,166 @@
+import os
+import re
+import requests
+import sys
+from google.cloud import bigquery
+from github import Github
+
+# Set up BigQuery client
+project_id = "idc-external-025"
+client = bigquery.Client(project=project_id)
+owner='ImagingDataCommons'
+# Set up GitHub client
+g = Github(os.environ["GITHUB_TOKEN"])
+repo = g.get_repo(f"{owner}/idc-index")
+
+def extract_index_version(file_path):
+    with open(file_path, "r") as file:
+        for line in file:
+            if "def get_idc_version(self):" in line:
+                return int(re.findall(r"v(\d+)", next(file))[0])
+
+def update_index_version(file_path, latest_idc_release_version):
+    with open(file_path, "r") as file:
+        lines = file.readlines()
+
+    with open(file_path, "w") as file:
+        for line in lines:
+            if "def get_idc_version(self):" in line:
+                line = re.sub(r"v(\d+)", f"v{latest_idc_release_version}", line)
+            file.write(line)
+
+def execute_sql_query(sql_query):
+    df = client.query(sql_query).to_dataframe()
+    return df
+
+def update_sql_query(file_path, current_index_version, latest_idc_release_version):
+    with open(file_path, "r") as file:
+        sql_query = file.read()
+
+    if current_index_version < latest_idc_release_version:
+        modified_sql_query = sql_query.replace(
+            f"idc_v{current_index_version}", f"idc_v{latest_idc_release_version}"
+        )
+
+        df = execute_sql_query(modified_sql_query)
+        csv_file_name = f"{os.path.basename(file_path).split('.')[0]}.csv.zip"
+        df.to_csv(csv_file_name, compression='gzip', escapechar="\\")
+
+        with open(file_path, "w") as file:
+            file.write(modified_sql_query)
+    else:
+        raise ValueError('Current version is not less than the latest version')
```

Comment:
Typically, exceptions are reserved for unexpected situations where it is not clear what needs to be done. In this case, it will be most common situation that the version did not change. The function should just return something different. Also, here you are making the function unnecessarily complex, and its name is not representative of what it does. I would have a separate function that detects whether version changed, and updates only if it did. This condition should be checked before the function, and if the check is negative, no need to call it: if current_index_version < latest_idc_release_version:.